### PR TITLE
Fixed knife vault show -F json and knife vault list -F json don't always output valid JSON

### PR DESCRIPTION
**Pull request by sanga1794**
## Description
Fixed knife vault show -F json and knife vault list -F json don't always output valid JSON

## Related Issue
Fixed: https://github.com/chef/chef-vault/issues/313

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/knife/vault_list.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def run
         bags.each_key do |bagname|
           vaultbags.push(bagname) if bag_is_vault?(bagname)
         end
-        output vaultbags.join("\n")
```
Comment (tas50) [inline lib/chef/knife/vault_list.rb:38]: Isn't this going to result in hard to read output on the CLI?

Comment (sanga1794) [reply lib/chef/knife/vault_list.rb:38]: Hi @tas50, nothing impacted on the CLI's output, but if we use join it is causing issues for JSON format, as posted in the issue.

File: lib/chef/knife/vault_show.rb [hunk 2]
```diff
@@ -90,9 +90,7 @@ def print_values(vault, item, values)
       def print_keys(vault)
         if bag_is_vault?(vault)
           bag = Chef::DataBag.load(vault)
-          split_vault_keys(bag)[1].each do |item|
-            output item
-          end
+          output split_vault_keys(bag)[1]
```
Comment (tas50) [inline lib/chef/knife/vault_show.rb:93]: If there's more than 1 key this will only show a single key. Is that intentional?

Comment (sanga1794) [reply lib/chef/knife/vault_show.rb:93]: Hi @tas50,yes it is intentional as split_vault_keys method return array of keys and items https://github.com/chef/chef-vault/blob/9e9d28e6dd8c43c32c65d1b69b46ed685a67f504/lib/chef/knife/vault_base.rb#L81.

And regarding the change made in PR, below are some o/ps before change and after change:

**Before Change:**
```
knife vault show val_passwords
   root
   root1
knife vault show val_passwords -F json
   "root"
   "root1"
knife vault show val_passwords -F yaml
	--- root
	...
	--- root1
	...
```

**After Change:**
```
knife vault show val_passwords
	root
	root1
knife vault show val_passwords -F json
	[
  	"root",
  	"root1"
	]
knife vault show val_passwords -F yaml
	---
	- root
	- root1
```

**Review by lamont-granquist — APPROVED** [review]

**Pull request merged**